### ISSUE TITLE
ML.NET 1.0.0.0 cannont read component 'HashTransform' of the model, because the model is too new.

### ISSUE THREAD

**Issue by wmundstock** [dotnet/machinelearning#5357]
### System information

- **OS:** Windows 10
- **Microsoft.Net Framework:** 4.7.2 
- **Model Builder:** 16.1.1.2041102
- **Visual Studio:** 16.6.3
- **ML.NET (Nuget packages installed):** tried 1.5.1 and 1.5.0

### Issue

- I have a **working** ML.Net app built with .Net framework 4.7, that consumes a model generated with Model Builder. It was working really well until I generated a new model file with an extra "feature". 
1. Re-Trained ML.NET model.
2. Generated the Model file and Input/output classes
3. Copied the ModelInput, ModelOutput and ModelConsume code to my application. 
4. Build and run.

- When I try to load the model it throws an exception **"ML.NET 1.0.0.0 cannont read component 'HashTransform' of the model, because the model is too new."**. Refer to exception details below.

**Note** that the *cannont* is not my typo, thats how the message shows.

- Given that I updated the model files and the code files required it should work well. If I change the model file and load the old zip model it works well.

### Source code / logs

`ITransformer mlModel = mlContext.Model.Load(MLNetModelPath, out var modelInputSchema);`

The line above throws the following exception:

**Message:** Cause: ML.NET 1.0.0.0 cannont read component 'HashTransform' of the model, because the model is too new.
Suggestion: Make sure the model is trained with ML.NET 1.0.0.0 or older.
Debug details: Maximum expected version 65540, got 65538.
**Source:** Microsoft.ML.Core
**Stack trace:**    at Microsoft.ML.ModelHeader.CheckVersionInfo(ModelHeader& header, VersionInfo ver)
   at Microsoft.ML.ModelLoadContext.CheckAtModel(VersionInfo ver)
   at Microsoft.ML.Transforms.HashingTransformer.Create(IHostEnvironment env, ModelLoadContext ctx)

### Notes
I really don't know what could have changed. Maybe Model builder auto updated (auto update is enabled on this extension)??? I really did not change anything related to ML.Net versions.

Things I already tried:
- Changed version of the nuget package for ML.Net (1.5.1 and 1.5.0) - no luck.
- Reinstalled model builder - no luck.
- Re-trained the same model (without the extra feature/column) - No luck.
- If I run the console app that is generated when I click generate code, that app is able to run. I suspect its because its .Net Core 3.1.

Thank you!

**Comment by harishsk** [comment]
Hi @wmundstock Your error message reads:
"ML.NET 1.0.0.0 cannont read component 'HashTransform' of the model, because the model is too new"

Is it the case that you added new feature, retrained the model and are using the retrained model with your old application?

If so, in this case this error is expected because in a recent version of ML.NET the hashing algorithm for the HashingTransformer was revised and a retrained model will use the newer hashing algorithm and the model version gets updated. So if you try to use the newer model on your older code, it will throw an error because it doesn't support this newer version.

**Comment by harishsk** [comment]
If it is an option for you, please consider updating your application version also to the same version that you retrained the model with.

**Comment by wmundstock** [comment]
Hi All. 

I was able to resolve the issue by removing all the Nuget packages and reinstalling only the absolute necessary. 

Again my app was .net 4.7 and it is a VSTO outlook addin. Which means it needed a few other references. When I added the references, for some reason it created a problem with the binding redirects, throwing an error Could not load file or assembly System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0.

During my previous debugs I've ended up adding more stuff than really needed, and by removing everything I went back to that exception, which is easily fixable by adding it as a Nuget package System.Runtime.CompilerServices.Unsafe version 4.7.1.

I suppose there were other ways to manually set the binding but that did the trick. Another alternative, from my reading, would be to go .Net Core, but don't think that is an option for me because the app was a VSTO addin.

Thanks, and hoping this may help someone someday.
